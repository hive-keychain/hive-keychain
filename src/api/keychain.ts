import axios from 'axios';

const KeychainApi = axios.create({
  baseURL:
    process.env.KEYCHAIN_API_DEV === 'true'
      ? 'http://localhost:3000/'
      : 'https://api.hive-keychain.com/',
});

export default KeychainApi;
