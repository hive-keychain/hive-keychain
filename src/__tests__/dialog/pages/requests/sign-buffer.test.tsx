import { KeychainKeyTypes } from '@interfaces/keychain.interface';
import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import SignBuffer from 'src/dialog/pages/requests/sign-buffer';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import mk from 'src/__tests__/utils-for-testing/data/mk';
import requestSignBuffer from 'src/__tests__/utils-for-testing/data/props/dialog/request-signBuffer';
import objects from 'src/__tests__/utils-for-testing/helpers/objects';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestSignBuffer } from 'src/__tests__/utils-for-testing/types/props-types';
describe('sign-buffer tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestSignBuffer;
  methods.config();
  it('Must show SignBuffer dialog(AnonymousRequest, known user, canwhitelist)', async () => {
    const clonedProps = objects.clone(props) as PropsRequestSignBuffer;
    delete clonedProps.accounts;
    delete clonedProps.data.title;
    clonedProps.data.username = mk.user.one;
    clonedProps.data.method = KeychainKeyTypes.posting;
    const { asFragment } = render(<SignBuffer {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot(
      'Request SignBuffer AnonymousRequest, known user, canwhitelist',
    );
  });
  it('Must show SignBuffer dialog(unknown, long message, no whitelisting)', async () => {
    const clonedProps = objects.clone(props) as PropsRequestSignBuffer;
    clonedProps.accounts = [mk.user.one, mk.user.two];
    delete clonedProps.data.username;
    clonedProps.data.method = KeychainKeyTypes.active;
    clonedProps.data.message = 'paracutirimicuaro '.repeat(80);
    const { asFragment } = render(<SignBuffer {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot(
      'Request SignBuffer unknown, long message, no whitelisting)',
    );
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<SignBuffer {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    render(<SignBuffer {...{ ...props, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    render(<SignBuffer {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
