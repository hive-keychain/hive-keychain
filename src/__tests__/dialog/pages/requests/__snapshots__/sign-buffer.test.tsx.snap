// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign-buffer tests:
 Must show SignBuffer dialog(AnonymousRequest, known user, canwhitelist): Request SignBuffer AnonymousRequest, known user, canwhitelist 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Sign Message
      </h2>
      <div
        aria-label="operation_header"
        class="operation_header "
      >
        The website domain would like to sign a message using the posting key for the account: @keychain.tests
      </div>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Message
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      />
    </div>
    <div
      class="operation_footer "
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      >
        <div
          class="checkbox-container align-center"
        >
          <div
            class="checkbox-and-label"
          >
            <div
              class="pretty p-default"
            >
              <input
                aria-checked="false"
                id="checkbox-set-keep-inner-input"
                type="checkbox"
                value=""
              />
              <div
                class="state"
              >
                <label
                  for="checkbox-set-keep-inner-input"
                />
              </div>
            </div>
            <div
              aria-label="checkbox-set-keep"
              class="label"
            >
              Do not prompt again to verify keys for the @keychain.tests account on domain
            </div>
          </div>
        </div>
      </div>
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`sign-buffer tests:
 Must show SignBuffer dialog(unknown, long message, no whitelisting): Request SignBuffer unknown, long message, no whitelisting) 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        title
      </h2>
      <div
        aria-label="operation_header"
        class="operation_header "
      >
        The website domain would like to sign a message using the active key for the account: @keychain.tests
      </div>
      <div
        class="select-account-section"
      >
        <div>
          <div
            aria-expanded="false"
            aria-label="Dropdown select"
            class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
            color="#0074D9"
            direction="ltr"
            tabindex="0"
          >
            <div
              class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
            >
              <div
                aria-label="selected-account-panel"
                class="selected-account-panel"
              >
                <div
                  aria-label="selected-account-name"
                  class="selected-account-name"
                >
                  keychain.tests
                </div>
              </div>
            </div>
            <div
              class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
              color="#0074D9"
              rotate="1"
              tabindex="-1"
            >
              <svg
                fill="currentColor"
                viewBox="0 0 40 40"
              >
                <path
                  d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Message
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicuaro paracutirimicu ...
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
